### DISCUSSION TITLE
Pagination plugin not working on the homepage of the website

### DISCUSSION THREAD

**Discussion by adamdjbrett** [mmistakes/minimal-mistakes#3292] [Q&A] [answered]
### What happened?

- Minimal Mistakes version: master
- Ruby gem or remote theme version: remote
- Jekyll version: Github Pages
- Git repository URL:  https://github.com/adamdjbrett/doctrineofdiscovery.org
- Hosted on GitHub Pages (if yes provide URL to site): doctrineofdiscovery.org
- Operating system: MacOS Catalina 10.15.7

### Expected behavior

on the [doctrienofdiscovery.org](https://doctrienofdiscovery.org) I expect to see the feature row and most recent posts. However the posts do not display the most recent post first.

### Steps to reproduce the behavior

using the layout splash combine feature row and pagination

### Other

I love Minimal Mistakes and I have been running it as master on [doctrineofdiscovery.org](https://doctrineofdiscovery.org/). However I noticed that the **Recent Posts** block on the homepage is not longer sorting by most recent post.
The code for recent posts works and runs on /home3/ fine and shows the most recent but that doesn't happen on the homepage and I am trying to figure what I did wrong or what I need to do to fix it. Any advice or suggestions would be greatly appreciated.

my homepage is running
```
layout: splash
```
and using both the 
```
{% include feature_row %}
```
and
```


<h3 class="archive__subtitle">{{ site.data.ui-text[site.locale].recent_posts | default: "Recent Posts" }}</h3>

{% if paginator %}
  {% assign posts = paginator.posts %}
{% else %}
  {% assign posts = site.posts %}
{% endif %}

{% assign entries_layout = page.entries_layout | default: 'list' %}
<div class="entries-{{ entries_layout }}">
  {% for post in posts %}
    {% include archive-single.html type=entries_layout %}
  {% endfor %}
</div>

{% include paginator.html %}
```

**Comment by mmistakes** [comment, marked as answer]
Jekyll's pagination plugin is extremely picky and only works correctly in certain situations.

1. You technically have two "home" pages... [`index.html`](https://github.com/adamdjbrett/doctrineofdiscovery.org/blob/master/index.html) and [`home3.html`](https://github.com/adamdjbrett/doctrineofdiscovery.org#:~:text=3%20years%20ago-,home3.html,-updating%20homepage)... remove `home3.html`
2. Remove `permalink: /` from `index.html`. It's not needed since it's already in the root, and by setting it as `/` it's confusing the paginator.
3. Remove all of the Liquid you added after `{% include feature_row %}`. It's not needed since you're already using `layout: home` on that page which has it. This is causing a duplication of "recent posts" on that page.
4. You may want to remove `/_pages/LICENSE.md` as well since it doesn't have any front matter and is outputting an extra `index.html` in `_pages` which may or may not be trouble for Jekyll's paginator.

After doing all of the above the home page should have your header, feature row, and recent posts showing properly. 

![127 0 0 1_4000_](https://user-images.githubusercontent.com/1376749/146832091-b96ea97e-4457-4e64-ae33-a8a08c356794.png)

If you want to use the wider `splash` layout I would suggest copying [`/_layouts/splash.html`](splash.html) from this repo, to your repo and rename it to something else. 

For example rename it to `splash-home.html`, and modify it like so to include the pagination stuff:

<details>
<summary>splash-home.html</summary>

```html
---
layout: default
---

{% if page.header.overlay_color or page.header.overlay_image or page.header.image %}
  {% include page__hero.html %}
{% elsif page.header.video.id and page.header.video.provider %}
  {% include page__hero_video.html %}
{% endif %}

<div id="main" role="main">
  <article class="splash" itemscope itemtype="https://schema.org/CreativeWork">
    {% if page.title %}<meta itemprop="headline" content="{{ page.title | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
    {% if page.excerpt %}<meta itemprop="description" content="{{ page.excerpt | markdownify | strip_html | strip_newlines | escape_once }}">{% endif %}
    {% if page.date %}<meta itemprop="datePublished" content="{{ page.date | date_to_xmlschema }}">{% endif %}
    {% if page.last_modified_at %}<meta itemprop="dateModified" content="{{ page.last_modified_at | date_to_xmlschema }}">{% endif %}

    <section class="page__content" itemprop="text">
      {{ content }}
    </section>
  </article>

  <h3 class="archive__subtitle">{{ site.data.ui-text[site.locale].recent_posts | default: "Recent Posts" }}</h3>
  {% if paginator %}
    {% assign posts = paginator.posts %}
  {% else %}
    {% assign posts = site.posts %}
  {% endif %}

  {% assign entries_layout = page.entries_layout | default: 'list' %}
  <div class="entries-{{ entries_layout }}">
    {% for post in posts %}
      {% include archive-single.html type=entries_layout %}
    {% endfor %}
  </div>

  {% include paginator.html %}
</div>

```
</details>

Then use `layout: splash-home` in `index.html` and you should get something that looks like this:

![127 0 0 1_4000_ (1)](https://user-images.githubusercontent.com/1376749/146832912-15c5b787-10eb-4fb9-b9e1-540637e74dd4.png)

**Comment by adamdjbrett** [comment]
Thank you so much for not only answering my question by providing such an elegant solution. I really appreciate your response and time. Thanks.
